src/algebra/ring/divisibility.lean

```diff
@@ -93,6 +93,22 @@ begin
     exact eq_add_of_sub_eq rfl }
 end
 
+theorem dvd_sub_iff_left (h : a ∣ c) : a ∣ b ↔ a ∣ b - c :=
+⟨λh₂, dvd_sub h₂ h, λH, by have t := dvd_add H h; rwa sub_add_cancel at t⟩
+
+theorem dvd_sub_iff_right (h : a ∣ b) : a ∣ c ↔ a ∣ b - c :=
+⟨λh₂, dvd_sub h h₂, λH, by have t := dvd_sub h H; rwa [← sub_add, sub_self, zero_add] at t⟩
+
+/-- If an element a divides another element c in a commutative ring, a divides the difference
+  of another element b with c iff a divides b. -/
+theorem dvd_sub_left (h : a ∣ c) : a ∣ b - c ↔ a ∣ b :=
+(dvd_sub_iff_left h).symm
+
+/-- If an element a divides another element b in a commutative ring, a divides the difference of b
+  and another element c iff a divides c. -/
+theorem dvd_sub_right (h : a ∣ b) : a ∣ b - c ↔ a ∣ c :=
+(dvd_sub_iff_right h).symm
```

Comment:
I guess you should remove the first two and keep the last two because the RHS is simpler than LHS, even though you don't mark them as simp lemmas.

Reply:
Yeah I copy pasted the entire api from the `add` versions, and I'd like it if they stay symmetric to make it easy to replace/copy proofs with add into ones with sub

Reply:
It looks like the add versions of the last two aren't used much in mathlib, but the first two are, so I'll have a think about what is the most natural way later

Reply:
Is there any updates on this? I think this is good to go, but in case you are still thinking about it I'll mark the PR as `awaiting-author` for now.